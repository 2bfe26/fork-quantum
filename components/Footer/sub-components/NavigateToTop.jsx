import React from 'react';
import styled from 'styled-components';
import { Container, Col } from '../../Grid';

const TopButtonHolder = styled(Col)`{
  text-align: right;
  padding: 10px;
  a{
    text-decoration: none;
    color: ${props => props.theme.primaryColor};
    font-family: ${props => props.theme.fontFamily};
    font-size: ${props => props.theme.fontSize};
    &::before {
      content: '\\25b2';
      padding-left: 0.5em;
    }
  }
}`;

const NavigateToTop = () => (
  <Container hide={['desktop', 'tablet', 'phone']}>
    <TopButtonHolder desktop={12}>
      <a href="#top">Topo</a>
    </TopButtonHolder>
  </Container>
);

export default NavigateToTop;
