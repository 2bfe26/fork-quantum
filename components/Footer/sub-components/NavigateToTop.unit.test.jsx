import React from 'react';
import renderer from 'react-test-renderer';
import NavigateToTop from './NavigateToTop';

describe('NavigateToTop component', () => {
  it('Should match the snapshot', () => {
    const tree = renderer.create(<NavigateToTop />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
