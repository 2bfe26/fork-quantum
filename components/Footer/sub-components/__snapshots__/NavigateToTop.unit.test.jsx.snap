// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigateToTop component Should match the snapshot 1`] = `
<div
  className="sc-bdVaJa dTrXBa"
>
  <div
    className="sc-ifAKCX vsVjH sc-htpNat cNhtKB"
  >
    <a
      href="#top"
    >
      Topo
    </a>
  </div>
</div>
`;
